app/models/permit_application.rb

```diff
@@ -495,41 +409,31 @@ def step_code_requirements
   def energy_step_code_required?
     custom_requirements = step_code_requirements.customizations
 
-    custom_requirements.empty? ||
-      custom_requirements.any? do |r|
-        r.energy_step_required || r.zero_carbon_step_required
-      end
+    custom_requirements.empty? || custom_requirements.any? { |r| r.energy_step_required || r.zero_carbon_step_required }
   end
 
   def self.count_by_jurisdiction_and_status
-    Jurisdiction.all.map do |j|
-      row =
-        PermitApplication
-          .with_submitter_role
-          .where(jurisdiction: j)
-          .group(:status)
-          .count
-
-      {
-        "draft" => (row["new_draft"] || 0) + (row["revisions_requested"] || 0), # nil defaults to 0
-        "submitted" => (row["newly_submitted"] || 0) + (row["resubmitted"] || 0)
-      }.merge({ name: j.qualified_name })
-    end
-  end
+    # Perform a single query to aggregate counts grouped by jurisdiction_id and status
+    counts =
+      PermitApplication.joins(:submitter).where(users: { role: "submitter" }).group(:jurisdiction_id, :status).count
```

Comment:
should we filter by submitter role? Can't a reviewer submit a permit application too in our app?